### DISCUSSION TITLE
Installation Process

### DISCUSSION THREAD

**Discussion by philippulus** [lolo101/MsgViewer#281] [Ideas]
Please kindly include the installation process in the code description, also updating info about java installation and the necessity to purge / remove any previous java installs and clean / remove existing MsgViewer folders in order for Maven install to run properly without errors.

$ sudo apt install openjdk-22-jdk
$ sudo apt install maven
$ sudo apt install git
$ git clone https://github.com/lolo101/MsgViewer.git
$ cd MsgViewer
$ ./mvnw package

**Comment by lolo101** [comment]
Hi @philippulus and thank you for the suggestion

The procedure currently described in the readme is essentially:
- Have JDK 17+ installed (here I cannot be too specific, many users do not use `apt` to manage their packages)
- git clone https://github.com/lolo101/MsgViewer.git
- cd MsgViewer
- ./mvnw package

I only think about one implicit that I can make explicit:
- git needs to be installed

However, it is unnecessary to install Maven thanks to the embedded Maven wrapper.
It is also unnecessary to purge any previous JDK installation when upgrading the JDK (you need to know how to manage different Java versions though)

There is also another implicit but I'm not convinced it is necessary to make it explicit:
- The instructions are for the very first build only. For subsequent builds, it is unnecessary to install again git, the JDK, and to clone the repository.

What do you think?

**Comment by philippulus** [comment]
I was fixated on Ubuntu installation, sorry. 
It's quite tricky to get the complete installation procedure right and I have been struggling with this so many times.
This is the reason I suggested this.
